packages/flutter/test/material/menu_bar_theme_test.dart

```diff
@@ -0,0 +1,324 @@
+
+  const TestMenu(this.label);
+  final String label;
+}
+
+List<Widget> createTestMenus({
+  void Function(TestMenu)? onPressed,
+  void Function(TestMenu)? onOpen,
+  void Function(TestMenu)? onClose,
+  Map<TestMenu, MenuSerializableShortcut> shortcuts = const <TestMenu, MenuSerializableShortcut>{},
+  bool includeStandard = false,
+  Color? itemOverlay,
+  Color? itemBackground,
+  Color? itemForeground,
+  EdgeInsetsDirectional? itemPadding,
+  Color? menuBackground,
+  EdgeInsetsDirectional? menuPadding,
+  OutlinedBorder? menuShape,
+  double? menuElevation,
+  OutlinedBorder? itemShape,
+}) {
+  final MenuStyle menuStyle = MenuStyle(
+    padding: menuPadding != null ? MaterialStatePropertyAll<EdgeInsetsGeometry>(menuPadding) : null,
+    backgroundColor: menuBackground != null ? MaterialStatePropertyAll<Color>(menuBackground) : null,
+    elevation: menuElevation != null ? MaterialStatePropertyAll<double>(menuElevation) : null,
+    shape: menuShape != null ? MaterialStatePropertyAll<OutlinedBorder>(menuShape) : null,
+  );
+  final ButtonStyle itemStyle = ButtonStyle(
+    padding: itemPadding != null ? MaterialStatePropertyAll<EdgeInsetsGeometry>(itemPadding) : null,
+    shape: itemShape != null ? MaterialStatePropertyAll<OutlinedBorder>(itemShape) : null,
+    foregroundColor: itemForeground != null ? MaterialStatePropertyAll<Color>(itemForeground) : null,
+    backgroundColor: itemBackground != null ? MaterialStatePropertyAll<Color>(itemBackground) : null,
+    overlayColor: itemOverlay != null ? MaterialStatePropertyAll<Color>(itemOverlay) : null,
+  );
+  final List<Widget> result = <Widget>[
+    MenuButton(
+      onOpen: onOpen != null ? () => onOpen(TestMenu.mainMenu0) : null,
+      onClose: onClose != null ? () => onClose(TestMenu.mainMenu0) : null,
+      menuChildren: <Widget>[
+        MenuItemButton(
+          onPressed: onPressed != null ? () => onPressed(TestMenu.subMenu00) : null,
+          shortcut: shortcuts[TestMenu.subMenu00],
+          child: Text(TestMenu.subMenu00.label),
+        ),
+      ],
+      child: Text(TestMenu.mainMenu0.label),
+    ),
+    MenuButton(
+      onOpen: onOpen != null ? () => onOpen(TestMenu.mainMenu1) : null,
+      onClose: onClose != null ? () => onClose(TestMenu.mainMenu1) : null,
+      menuStyle: menuStyle,
+      menuChildren: <Widget>[
+        MenuItemButton(
+          onPressed: onPressed != null ? () => onPressed(TestMenu.subMenu10) : null,
+          shortcut: shortcuts[TestMenu.subMenu10],
+          style: itemStyle,
+          child: Text(TestMenu.subMenu10.label),
+        ),
+        MenuButton(
+          onOpen: onOpen != null ? () => onOpen(TestMenu.subMenu11) : null,
+          onClose: onClose != null ? () => onClose(TestMenu.subMenu11) : null,
+          menuChildren: <Widget>[
+            MenuItemButton(
+              onPressed: onPressed != null ? () => onPressed(TestMenu.subSubMenu100) : null,
+              shortcut: shortcuts[TestMenu.subSubMenu100],
```

Comment:
I'm not sure it is worth it, consider renaming `subSubMenu100` (and similar) to `subSubMenu110`